src/System.Private.CoreLib/src/System/StartupHookProvider.cs

```diff
@@ -0,0 +1,120 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using System.Runtime.Loader;
+
+namespace System
+{
+    internal static class StartupHookProvider
+    {
+        private const string StartupHookTypeName = "StartupHook";
+        private const string InitializeMethodName = "Initialize";
+
+        // Parse a string specifying a list of assemblies and types
+        // containing a startup hook, and call each hook in turn.
+        private static void ProcessStartupHooks()
+        {
+            string startupHooksVariable = (string)AppContext.GetData("STARTUP_HOOKS");
+            if (startupHooksVariable == null)
+            {
+                return;
+            }
+
+            // Parse startup hooks variable
+            string[] startupHooks = startupHooksVariable.Split(Path.PathSeparator);
+            foreach (string startupHook in startupHooks)
+            {
+                if (String.IsNullOrEmpty(startupHook))
+                {
+                    throw new ArgumentException(SR.Argument_InvalidStartupHookSyntax);
+                }
+                if (PathInternal.IsPartiallyQualified(startupHook))
+                {
+                    throw new ArgumentException(SR.Argument_AbsolutePathRequired);
```

Comment:
This will just say `Absolute path information is required.` without any indication of what the path was or where it came from.

Reply:
The stack trace will indicate that the exception came from the startup hook path. This is the same exception we throw elsewhere from SPC for non-absolute path info. Sometimes we also include the name of the variable containing the path, but that wouldn't help in this case since it's a local that could be any of the hooks.

Reply:
OK thanks

Reply:
I may make a pass in master in future to add the path (if it exists) to all uses of Argument_AbsolutePathRequired... 😄